src/uhh/CMakeLists.txt

```diff
@@ -2,6 +2,8 @@ add_library(uhh-class uhh.cpp)
 
 list(APPEND EXTRA_LIBS stdc++fs)
 list(APPEND EXTRA_LIBS git2::git2)
+list(APPEND EXTRA_LIBS config-class)
+list(APPEND EXTRA_LIBS uhh-git)
```

Comment:
these were missing and were causing my builds on fail locally.

IMO we are going a little overboard with each class being it's own library, keeping uhh specific classes in one library might simplify the builds. Ninja takes care of incremental builds so we don't have to implement then ourselves.

Reply:
Its simply the only way I know how to cmake :) 

Remember, i am a complete and utter rookie when it comes to cmake, programming, walking, and other life activities.